libr/main/radare2.c

```diff
@@ -1221,8 +1221,15 @@ R_API int r_main_radare2(int argc, char **argv) {
 			if (compute_hashes && iod) {
 				// TODO: recall with limit=0 ?
 				ut64 limit = r_config_get_i (r->config, "bin.hashlimit");
-				(void)r_bin_file_hash (r->bin, limit, iod->name, NULL);
-				//eprintf ("WARNING: File hash not calculated\n");
+				RList *file_hashes = r_bin_file_hash (r->bin, limit);
+				if (file_hashes) {
+					if (r->bin->cur->o->info->file_hashes) {
+						r_list_join (r->bin->cur->o->info->file_hashes, file_hashes);
+						free (file_hashes);
```

Comment:
this probably causes a Use After Free, because it also frees each element in file_hashes, which are now owned by info->file_hashes.

Reply:
From my understanding, `r_list_join` join the list by pointing the tail of the first list to the head of the second list, so I would need to free the struct RList after that and there is another `r_list_free` which free the elements.

Reply:
Oooops sorry I misread. I thought you were calling r_list_free, but you are correctly calling free!